tee_launcher/test_launcher_config.py

```diff
@@ -372,4 +386,80 @@ def test_ld_preload_in_ports2():
     assert "LD_PRELOAD" not in docker_cmd  # Ensure LD_PRELOAD is NOT in the command
 
 
-# Additional tests could go here to check other edge cases
+def test_json_key_matches_node():
+    """
+    Ensure the JSON key used by the launcher to read approved image hashes
+    stays aligned with the MPC node implementation.
+    mpc/crates/node/src/tee/allowed_image_hashes_watcher.rs -> JSON_KEY_APPROVED_HASHES
+
+    If this test fails, it means the launcher and MPC node are using different
+    JSON field names, which would break MPC hash propagation.
+    """
+    assert launcher.JSON_KEY_APPROVED_HASHES == "approved_hashes"
+
+def test_override_present(monkeypatch):
+    override_value = "sha256:" + "a"*64
+    approved = [override_value, "sha256:" + "b"*64]
+
+    fake_json = make_digest_json(approved)
+
+    monkeypatch.setattr("tee_launcher.launcher.os.path.isfile", lambda _: True)
+    monkeypatch.setattr("builtins.open", mock_open(read_data=fake_json))
```

Comment:
I was thinking to use this the other day, didn't think would see it so soon, nice!